nautobot/generate_secret_key.py

```diff
@@ -1,8 +1,15 @@
 #!/usr/bin/env python
 # This script will generate a random 50-character string suitable for use as a SECRET_KEY.
-from random import SystemRandom
+# shamelessly pulled from https://docs.python.org/3/library/secrets.html#recipes-and-best-practices
+import string
+import secrets
 
-choice = SystemRandom().choice
-
-charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*(-_=+)"
-print("".join(choice(charset) for _ in range(50)))
+length = 50
+alphabet = string.ascii_letters + string.digits + "!@#$%^&*(-_=+)"  # string.punctuation would be an option
+while True:
+    password = ''.join(secrets.choice(alphabet) for i in range(length))
+    if (any(c.islower() for c in password)
+            and any(c.isupper() for c in password)
+            and sum(c.isdigit() for c in password) >= 3):
```

Comment:
Looking at the Django code, the actual check that it performs at startup to confirm that the SECRET_KEY is good is a `len(set(password)) >= 5` to confirm that there's a decent number of unique characters in the key. Would it make sense to add that, or perhaps even replace this set of checks?

Reply:
The check ensures a minimum amount of lowercase, uppercase and digits in the generated string. 

From my point of view it serves two purposes:
1. Ensure a certain mix of character classes
2. Prevent generation of substandard secrets due to a broken PRNG implementation

Reply:
Sure